### ISSUE TITLE
bug: false positive of `no-unused-vars` for symbols used in TSDoc comments

### ISSUE THREAD

**Issue by KnorpelSenf** [denoland/deno_lint#1472]
### Lint Name

`no-unused-vars`

### Code Snippet

```ts
import type { Foo } from "./file.ts";

/**
 * Related to {@link Foo.bar | this property}
 */
const baz = 42;
```

### Expected Result

The symbol `Foo` is needed in order to make the TSDoc comment of `baz` valid, so it is used. I expect `deno_lint` to see this, and to exit with status code 0.

### Actual Result

`no-unused-vars` is reported

### Version

deno 2.6.5 (stable, release, x86_64-unknown-linux-gnu)
v8 14.2.231.17-rusty
typescript 5.9.2

**Comment by tobias47n9e** [comment]
Might be related to this issue: #1449

I can try to fix this.
